airbyte-cdk/python/airbyte_cdk/sources/declarative/types.py

```diff
@@ -16,43 +16,38 @@
 
 
 class Record(Mapping[str, Any]):
-    def __init__(self, data: dict, associated_slice: StreamSlice):
+    def __init__(self, data: Mapping[str, Any], associated_slice: Optional[StreamSlice]):
         self._data = data
         self._associated_slice = associated_slice
 
     @property
-    def data(self) -> dict:
+    def data(self) -> Mapping[str, Any]:
         return self._data
 
     @property
-    def associated_slice(self) -> StreamSlice:
+    def associated_slice(self) -> Optional[StreamSlice]:
         return self._associated_slice
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return repr(self._data)
 
-    def __setitem__(self, key: str, value: Any):
-        self._data[key] = value
-
-    def __getitem__(self, key: str):
+    def __getitem__(self, key: str) -> Any:
         return self._data[key]
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._data)
 
-    def __iter__(self):
+    def __iter__(self) -> Any:
         return iter(self._data)
 
-    def __contains__(self, item: str):
+    def __contains__(self, item: object) -> bool:
         return item in self._data
 
-    def __eq__(self, other):
-        if isinstance(other, dict):
```

Comment:
Tests were passing before because we allowed the comparison with a dict while those have two very different interfaces